### ISSUE TITLE
rabin2: shared object files reported as 'static'

### ISSUE THREAD

**Issue by armijnhemel** [radare/radare2#9331]
# This template is meant for bug reports, if you have a feature request, please be as descriptive as possible and delete the template

Make sure you are testing using the latest git version of radare2 before submitting any issue.

*If you would like to report a bug, please fill the template bellow*

### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Fedora 26, x86_64
| File format of the file you reverse (mandatory)      | ELF
| Architecture/bits of the file (mandatory)            | x86_64, MIPS
| r2 -v full output, **not truncated** (mandatory)         | radare2 2.4.0-git 17206 @ linux-x86-64 git.2.2.0-398-g6ba461f1c
commit: 6ba461f1c1c0e32fad16e2191676711da02865ad build: 2018-02-06__17:34:21

| r2 -V full output in a pastebin document (mandatory) | https://gist.github.com/armijnhemel/3d6cbc134530bbc6ff37eff07029404d

### Expected behavior

rabin2 -Ij should report that dynamically linked libraries are dynamically linked, not statically

### Actual behavior

rabin2 -Ij reports that dynamically linked libraries are static:

rabin2 -Ij /lib64/libncurses.so.6 | json_pp | grep static
      "static" : true,

readelf says:

readelf -Wa /lib64/libncurses.so.6 | grep NEEDED
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libdl.so.2]
 0x0000000000000001 (NEEDED)             Shared library: [libtinfo.so.6]
[libncurses.zip](https://github.com/radare/radare2/files/1700125/libncurses.zip)

### Steps to reproduce the behavior 
- Please share the binary if it is shareable by drag and dropping it here in a zip archive (mandatory)
- Use asciinema to describe the issue and share the link here (mandatory if u can't share the binary) 
- Use code markdown `CODE` to make your code visible
- Or even better, create a Pull-Request for the https://github.com/radare/radare2-regressions containing the test case examples can be found in the t/ folder see for example https://github.com/radare/radare2-regressions/blob/master/t/cmd_search#L7

### Additional Logs, screenshots, source-code,  configuration dump, ...

Drag and drop zip archives containing the Additional info here, don't use external services or link.

**Comment by armijnhemel** [comment]
Urgh, sorry for the messy bug report. The ZIP file that is included contains the stock Fedora 26 libncurses libraries. I will try to triage a bit.

**Comment by radare** [comment]
Thats an elf specific issue. Cant repro on dll or dylibs

**Comment by armijnhemel** [comment]
> Thats an elf specific issue. Cant repro on dll or dylibs

I'm sorry, but I am very confused by your answer. Is this a bug, or are you saying it is expected behaviour?

I went through the code and it looks like the only check that matters in radare2 is whether or not an interpreter section is found, which would always be true for every dynamically linked ELF library. I checked how for example the "file" program is doing this and it seems that they also check whether or not there is a dynamic section (PT_DYNAMIC).
